nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/ResultSetRecordSet.java

```diff
@@ -298,7 +272,67 @@ private DataType getDataType(final int sqlType, final ResultSet rs, final int co
         }
     }
 
-    private static DataType getArrayBaseType(final Array array) throws SQLException {
+    private DataType determineDataTypeToReturn(DataType dataType, boolean useLogicalTypes) {
+        RecordFieldType fieldType = dataType.getFieldType();
+        if (!useLogicalTypes
+                && (fieldType == RecordFieldType.DECIMAL
+                || fieldType == RecordFieldType.DATE
+                || fieldType == RecordFieldType.TIME
+                || fieldType == RecordFieldType.TIMESTAMP)) {
+            return RecordFieldType.STRING.getDataType();
+        } else {
+            return dataType;
+        }
+    }
+
+    private DataType getArrayDataType(ResultSet rs, int columnIndex, boolean useLogicalTypes) throws SQLException {
```

Comment:
Recommend adding `final` to these parameters:
```suggestion
    private DataType getArrayDataType(final ResultSet rs, final int columnIndex, final boolean useLogicalTypes) throws SQLException {
```